src/components/color_picker/color_stops/color_stop_thumb.tsx

```diff
@@ -0,0 +1,315 @@
+  const [stopIsInvalid, setStopIsInvalid] = useState(isStopInvalid(stop));
+  const [numberInputRef, setNumberInputRef] = useState();
+  const popoverRef = useRef<EuiPopover>(null);
+
+  useEffect(() => {
+    if (isPopoverOpen && popoverRef && popoverRef.current) {
+      popoverRef.current.positionPopoverFixed();
+    }
+  }, [stop]);
+
+  const getStopFromMouseLocationFn = (location: { x: number; y: number }) => {
+    // Guards against `null` ref in usage
+    return getStopFromMouseLocation(location, parentRef!, globalMin, globalMax);
+  };
+
+  const getPositionFromStopFn = (stop: ColorStop['stop']) => {
+    // Guards against `null` ref in usage
+    return getPositionFromStop(stop, parentRef!, globalMin, globalMax);
+  };
+
+  const openPopover = () => setIsPopoverOpen(true);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const handleOnRemove = () => {
+    if (onRemove) {
+      closePopover();
+      onRemove();
+    }
+  };
+
+  const handleColorChange = (value: ColorStop['color']) => {
+    setColorIsInvalid(isColorInvalid(value));
+    onChange({ stop, color: value });
+  };
+
+  const handleStopChange = (
+    value: ColorStop['stop'],
+    shouldRespectBoundaries: boolean = false
+  ) => {
+    const willBeInvalid = value > max || value < min;
+
+    if (shouldRespectBoundaries && willBeInvalid) {
+      if (value > max) {
+        value = max;
+      }
+      if (value < min) {
+        value = min;
+      }
+    }
+    setStopIsInvalid(isStopInvalid(value));
+    onChange({ stop: value, color });
+  };
+
+  const handleStopInputChange = (value: ColorStop['stop']) => {
+    const willBeInvalid = value > globalMax || value < globalMin;
+
+    if (willBeInvalid) {
+      if (value > globalMax) {
+        value = globalMax;
+      }
+      if (value < globalMin) {
+        value = globalMin;
+      }
+    }
+    setStopIsInvalid(isStopInvalid(value));
+    onChange({ stop: value, color });
+  };
+
+  const handlePointerChange = (
+    location: { x: number; y: number },
+    isFirstInteraction?: boolean
+  ) => {
+    if (isFirstInteraction) return; // Prevents change on the inital MouseDown event
+    if (parentRef == null) {
+      return;
+    }
+    const newStop = getStopFromMouseLocationFn(location);
+    handleStopChange(newStop, true);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+    switch (e.keyCode) {
+      case keyCodes.LEFT:
+        e.preventDefault();
+        handleStopChange(stop - 1, true);
+        break;
+
+      case keyCodes.RIGHT:
+        e.preventDefault();
+        handleStopChange(stop + 1, true);
+        break;
+    }
+  };
+
+  const [handleMouseDown, handleInteraction] = useMouseMove<HTMLButtonElement>(
+    handlePointerChange
+  );
+
+  return (
+    <EuiPopover
+      ref={popoverRef}
+      className="euiColorStopPopover"
+      anchorClassName="euiColorStopPopover__anchor"
+      panelPaddingSize="m"
```

Comment:
I'd make this `s` so all the margins match up

![image](https://user-images.githubusercontent.com/324519/65726324-aa02aa00-e069-11e9-8af0-814acbf58daa.png)
```suggestion
      panelPaddingSize="s"
```

Reply:
+1 on the change to `s`.